src/dbnode/persist/fs/retriever.go

```diff
@@ -431,48 +448,58 @@ func (r *blockRetriever) fetchBatch(
 		req.indexEntry = entry
 	}
 
-	sort.Sort(retrieveRequestByOffsetAsc(reqs))
+	sort.Sort(retrieveRequestByIndexEntryOffsetAsc(reqs))
 	tagDecoderPool := r.fsOpts.TagDecoderPool()
 
 	blockCachingEnabled := r.opts.CacheBlocksOnRetrieve() && r.nsCacheBlocksOnRetrieve
 
 	// Seek and execute all requests
 	for _, req := range reqs {
-		var (
-			data checked.Bytes
-			err  error
-		)
+		// Should always be a data request by this point.
```

Comment:
nit: maybe `switch` would be more readable here instead of the following 3 or 4 `ifs`?

Reply:
They're unrelated conditions though, so not great with `switch`